sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -669,9 +668,17 @@ public GenericDatumTransformer(
         String tableSchema,
         org.apache.avro.Schema writer) {
       this.parseFn = parseFn;
-      this.tableSchema =
-          Suppliers.memoize(
-              Suppliers.compose(new TableSchemaFunction(), Suppliers.ofInstance(tableSchema)));
+      this.tableSchema = new TableSchemaFunction().apply(tableSchema);
```

Comment:
We can remove TableSchemaFunction as well as this first constructor

Reply:
Thank you for this suggestion. It was needed to convert the JSON string constructor parameter into a TableSchema. I felt that since the method was public, we needed to still support it and thus needed this conversion using TableSchemaFunction. @Abacn in light of this context, do you agree?

Reply:
TableSchemaFunction is private and GenericDatumTransformer is package private. The public constructor just mean it can be called within the package

Reply:
Well, as of this PR itself the current change suffices and LGTM. This could be cleaned up / or leave it as is

Reply:
+1 to removing `TableSchemaFunction` and the constructor, it can't be accessed by users anyways. We can get the tableschema from json directly with `BigQueryHelpers.fromJsonString(tableschema, TableSchema.class)`